### PR TITLE
[manuf] Add SECRET2 and info flash secret configuration

### PR DESCRIPTION
**Pull request by moidx**
This change adds configuration of the root secret in the SECRET2 OTP partition as well as the silicon creator seed stored in info flash.
 
The test currently has some workarounds in place to implement OTP and flash_ctrl logic until we switch testutil libraries to use status_t. 

There is also a ft.c functional test target used to run the test case with a DEV lc stage bitstream.

Part of https://github.com/lowRISC/opentitan/issues/17393

### CODE REVIEW

File: sw/device/silicon_creator/manuf/BUILD [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+load("//rules:opentitan_test.bzl", "cw310_params", "opentitan_functest", "verilator_params")
+
+opentitan_functest(
+    name = "ft",
+    srcs = ["ft.c"],
+    cw310 = cw310_params(
+        bitstream = "//hw/bitstream:rom_otp_dev",
+        # TODO(lowRISC/opentitan#13603): Remove this "manual" tag when the
+        # bitstream target can fetch pre-spliced bitstream from GCP.
+        tags = ["manual"],
```
Comment (timothytrippel) [inline sw/device/silicon_creator/manuf/BUILD:14]: I don't think you need this anymore, we can splice in CI and locally now.

File: sw/device/silicon_creator/manuf/BUILD [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("//rules:opentitan_test.bzl", "cw310_params", "opentitan_functest", "verilator_params")
+
+opentitan_functest(
+    name = "ft",
+    srcs = ["ft.c"],
+    cw310 = cw310_params(
```
Comment (timothytrippel) [inline sw/device/silicon_creator/manuf/BUILD:10]: do you want to run this test with ROM only? or test ROM too? if ROM only. then you can add:
```
targets = [
    "cw310_rom",
    "verilator",
],
```

File: sw/device/silicon_creator/manuf/BUILD [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        bitstream = "//hw/bitstream:rom_otp_dev",
+        # TODO(lowRISC/opentitan#13603): Remove this "manual" tag when the
+        # bitstream target can fetch pre-spliced bitstream from GCP.
+        tags = ["manual"],
+    ),
+    verilator = verilator_params(
+        timeout = "long",
```
Comment (timothytrippel) [inline sw/device/silicon_creator/manuf/BUILD:17]: Want to add the OTP image here:
```
otp = "<bazel target>",
```
If running with ROM only, then also add:
```
    rom = "//sw/device/silicon_creator/rom:rom_with_fake_keys",
```

File: sw/device/silicon_creator/manuf/BUILD [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("//rules:opentitan_test.bzl", "cw310_params", "opentitan_functest", "verilator_params")
+
+opentitan_functest(
+    name = "ft",
```
Comment (timothytrippel) [inline sw/device/silicon_creator/manuf/BUILD:8]: nit: maybe a more descriptive name for the test? (can update later too)

File: sw/device/silicon_creator/manuf/ft.c [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+bool test_main(void) {
+  CHECK_STATUS_OK(peripheral_handles_init());
+
+  // TODO: This test is a temporary placeholder which will get replaced for an
+  // interactive test loop once we have host side instrumentation available.
+  CHECK_STATUS_OK(
+      provisioning_device_secrets_start(&flash_state, &lc_ctrl, &otp_ctrl));
+
```
Comment (timothytrippel) [inline sw/device/silicon_creator/manuf/ft.c:50]: going to call `provisioning_device_secrets_end()` here? or still adding stuff?

**Review by timothytrippel — APPROVED** [review]

File: sw/device/silicon_creator/manuf/ft.c (hunk 5 above)
Comment (moidx) [reply sw/device/silicon_creator/manuf/ft.c:50]: yeah, still adding stuff. I will add a device reset and then call the end function.

File: sw/device/silicon_creator/manuf/BUILD (hunk 4 above)
Comment (moidx) [reply sw/device/silicon_creator/manuf/BUILD:8]: should I move this under `manuf/lib/provisioning_functest.c`?

Comment (timothytrippel) [reply sw/device/silicon_creator/manuf/BUILD:8]: sgtm, that would be consistent with Si creator code

File: sw/device/silicon_creator/manuf/BUILD (hunk 1 above)
Comment (moidx) [reply sw/device/silicon_creator/manuf/BUILD]: Done.

File: sw/device/silicon_creator/manuf/BUILD (hunk 2 above)
Comment (moidx) [reply sw/device/silicon_creator/manuf/BUILD]: Choosing `cw310_rom` for now. Will add verilator once we have otp scaffolding.

File: sw/device/silicon_creator/manuf/BUILD (hunk 4 above)
Comment (moidx) [reply sw/device/silicon_creator/manuf/BUILD]: Done.

Comment (moidx) [reply sw/device/silicon_creator/manuf/BUILD]: Done.

**Pull request merged**